### ISSUE TITLE
Java agent fails to install

### ISSUE THREAD

**Issue by danieljimenez** [escapestudios-cookbooks/newrelic#193]
```
Chef::Mixin::Template::TemplateError
                         ------------------------------------
                         undefined method `agent_type' for nil:NilClass
                         
                         Resource Declaration:
                         ---------------------
                         # In /var/chef/cache/cookbooks/foreflight_server/recipes/deploy_instances.rb
                         
                          18:   newrelic_yml File.join(node[:resin][:home], 'ext-lib', 'newrelic.yml') do
                          19:     agent_type 'java'
                          20:     app_name "#{instance} (#{node[:foreflight][:environment][:name]})"
                          21:   end
                          22: 
                         
                         Compiled Resource:
                         ------------------
                         # Declared in /var/chef/cache/cookbooks/foreflight_server/recipes/deploy_instances.rb:18:in `block in from_file'
                         
                         newrelic_yml("/foreflight/resin/ext-lib/newrelic.yml") do
                           action :generate
                           retries 0
                           retry_delay 2
                           default_guard_interpreter :default
                           declared_type :newrelic_yml
                           cookbook_name "foreflight_server"
                           recipe_name "deploy_instances"
                           agent_type "java"
                           app_name "public (skylab)"
                           yml_path "/foreflight/resin/ext-lib/newrelic.yml"
                           template_cookbook "newrelic"
                           template_source "agent/newrelic.yml.erb"
                           license ""
                           loglevel "INFO"
                           transaction_tracer_enable true
                           transaction_tracer_record_sql true
                         end
                         
                         Template Context:
                         -----------------
                         on line #6
                           4: #
                           5: # This file configures the New Relic Agent.  New Relic monitors
                           6: # <%= @resource.agent_type %> applications with deep visibility and low overhead.  For more
                           7: # information, visit www.newrelic.com.
                           8: #
```

**Comment by djoos** [comment]
Hi @danieljimenez,

sorry for the delay in getting back to you!
Which version of the cookbook are you using exactly?

Additionally: would you be able to let me know why exactly you're using the newrelic_yml LWRP as opposed to making use of the newrelic_agent_java LWRP in order to get the Java agent installed?
Please have a look at [agent_java.rb recipe](https://github.com/escapestudios-cookbooks/newrelic/blob/master/test/fixtures/cookbooks/newrelic_lwrp_test/recipes/agent_java.rb) for a simple example.

Thanks in advance for your feedback!

Kind regards,
David

**Comment by matkam** [comment]
@djoos I'm having the same issue using newrelic_yml with version 2.12.3.

```
newrelic_yml "#{config_dir}/newrelic.yml" do
  agent_type 'ruby'
  enabled node['newrelic']['application_monitoring']['enabled']
  app_name node['group']['app_name']['newrelic']['app_name']
  logfile_path node['group']['app_name']['log_dir']
  owner deploy_user
  group unicorn_group
end
```

```
[2015-06-18T17:59:44+00:00] ERROR:

Chef::Mixin::Template::TemplateError (undefined method `agent_type' for nil:NilClass) on line #6:

  4: #
  5: # This file configures the New Relic Agent.  New Relic monitors
  6: # <%= @resource.agent_type %> applications with deep visibility and low overhead.  For more
  7: # information, visit www.newrelic.com.
  8: #
```

I'm using the newrelic_yml LWRP as opposed to the newrelic_agent_ruby LWRP because I'm running a rails app that already installs the newrelic gem and I just want a config file to be generated. This was working until recently, assuming there was a recent change.

**Comment by djoos** [comment]
Aha, ok - I think I know what is going on here.

Would you mind giving [bugfix/issue-193](https://github.com/escapestudios-cookbooks/newrelic/tree/bugfix/issue-193) a go?

Thanks in advance for your feedback!

Kind regards,
David

**Comment by matkam** [comment]
Closer, but:
```
Chef::Mixin::Template::TemplateError (undefined method `daemon_proxy_host' for Chef::Resource::NewrelicYml) on line #152:

150:   # proxy_user: username
151:   # proxy_password: password 
152:   <% unless @resource.daemon_proxy_host.nil? %>
153:   proxy_host: <%= @resource.daemon_proxy_host %>
154:   <% end %>
```

**Comment by djoos** [comment]
Could you give the latest from that branch another go?

Thanks in advance!

Kind regards,
David

**Comment by matkam** [comment]
Works thanks!  :+1:

**Comment by djoos** [comment]
Cool, let me ship that out... Thanks for your feedback!

**Comment by djoos** [comment]
Fixed in [2.12.4](https://github.com/escapestudios-cookbooks/newrelic/releases/tag/2.12.4)!

Kind regards,
David
